### PR TITLE
Integrate Echelon

### PR DESCRIPTION
**Pull request by fkorotkov**
cli-mark-2

### CODE REVIEW

File: internal/commands/run.go [hunk 1]
```diff
@@ -43,6 +46,9 @@ func run(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	if !verbose {
```
Comment (edigaryev) [inline internal/commands/run.go:49]: These two things (verbosity and interactivity) seem to be orthogonal: you can have both interactive and verbose output at the same time.

Take a look at `-1` argument of the [`ls`](https://man.openbsd.org/ls) command if you want an option to disable interactive output.

File: cmd/cirrus/main.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func main() {
-	// Set up signal interruptible context
+	// Set up signal interruptable context
```
Comment (edigaryev) [inline cmd/cirrus/main.go:12]: https://en.wiktionary.org/wiki/interruptible

File: go.sum [hunk 3]
```diff
@@ -22,6 +22,14 @@ github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghf
...
 github.com/cirruslabs/cirrus-ci-annotations v0.0.0-20200708182643-6441beff996d/go.mod h1:98qD7HLlBx5aNqWiCH80OTTqTTsbXT69wxnlnrnoL0E=
+github.com/cirruslabs/echelon v0.0.0-20200803173629-85f41761ba61 h1:/oV6FLQLFQ21k2tgbkgV56D4jfCJAopDcObMos+Zibc=
+github.com/cirruslabs/echelon v0.0.0-20200803173629-85f41761ba61/go.mod h1:c57rTdFNE3E34PRYuWXFYZyb+xDozdqS5Fw5TlM7wxg=
+github.com/cirruslabs/echelon v0.0.0-20200803185240-d7b99d045349 h1:ffYD8ueB/pVjBAAIcpaG/0FR5jvAG0Viro09JQx8oQQ=
+github.com/cirruslabs/echelon v0.0.0-20200803185240-d7b99d045349/go.mod h1:c57rTdFNE3E34PRYuWXFYZyb+xDozdqS5Fw5TlM7wxg=
+github.com/cirruslabs/echelon v0.0.0-20200803202207-2e5d9027b8c4 h1:cVJsDiaL/0KFvurL+uzw6KX5zaVrvky2JRlXVqX2u2M=
+github.com/cirruslabs/echelon v0.0.0-20200803202207-2e5d9027b8c4/go.mod h1:c57rTdFNE3E34PRYuWXFYZyb+xDozdqS5Fw5TlM7wxg=
+github.com/cirruslabs/echelon v0.0.0-20200803204805-aa942972b98a h1:dX07a1HRRke/WQBgj3S13eOnGqZmmQSF/fE5m8jhxTA=
+github.com/cirruslabs/echelon v0.0.0-20200803204805-aa942972b98a/go.mod h1:c57rTdFNE3E34PRYuWXFYZyb+xDozdqS5Fw5TlM7wxg=
```
Comment (edigaryev) [inline go.sum:32]: This clearly needs `go mod tidy`.

File: cmd/cirrus/main.go (hunk 2 above)
Comment (fkorotkov) [reply cmd/cirrus/main.go:12]: 🤷‍♂️ https://en.wiktionary.org/wiki/interruptable

IntelliJ's dictionary suggested it. Will revert. 😅

File: internal/commands/run.go (hunk 1 above)
Comment (fkorotkov) [reply internal/commands/run.go:49]: I don't see how it can play well with the incremental redrawing. I can try to add log levels to echelon and have description only show messages for a particular log level. Then we can try to replace logrus completely with echelon. 🤔 What do you think?

Comment (edigaryev) [reply internal/commands/run.go:49]: That's an interesting idea. Could the echelon library provide a logrus-compatible interface to simplify integrations?

This way the only thing we'll have to do here is to swap the logger.

Comment (fkorotkov) [reply internal/commands/run.go:49]: You still need to create the hierarchy but the `Debugf` and context things can be replicated. Will experiments with it. 👍

**Comment by fkorotkov** [comment]
Need to update Gif though. Will do it tomorrow.

**Comment by fkorotkov** [comment]
This is now ready for another round of review. Now Echelon supports interactive mode like on the GIF as well as a simple colored output for dump terminals like in CI.

**Comment by fkorotkov** [comment]
Hold on, will fix tests first. 😅

**Comment by fkorotkov** [comment]
Fixed tests 👌😅
